src/main/java/org/reactome/qa/doisuggester/Main.java

```diff
@@ -50,213 +56,201 @@ public static void main(String[] args) throws FileNotFoundException, IOException
-											{
-												// If 'Pathway' key doesn't exist, create it and then add to it.
-												Set<Map<GKInstance, GKInstance>> reaction2newIESet = new HashSet<>(Arrays.asList(reaction2newIE));
-												pathway2Reactions2NewIEs.put(parentPathway, reaction2newIESet);
-											}
-										}
-									}
-								}
-							}
-						}
+						pathway2Reactions2NewIEs = processPreexistingRLEs(pathway2Reactions2NewIEs, currentRLE, previousRLE);
 					}
+					
 				}
 			}
 
-			// For cases where this the first Release that a RlE has appeared (ie. fetching
-			// for the RlE in the previous DB returned null).
+			Map<GKInstance, Set<Map<GKInstance, GKInstance>>> pathwaysMap = processNewRLEs(newRLEs, pathway2Reactions2NewIEs);
 
-			// As before, iterate through all RlEs.
-			for (GKInstance newRLE : newRLEs)
-			{
-				// Get non-Reactome IEs from the revised, authored, and reviewed attributes.
-				List<GKInstance> newRLEInstanceEdits = getNonReactomeEventIEs(newRLE);
+			printOutput(pathwaysMap);
 
-				// Look through all parents of the RlE, trying to find the new instance edit in
-				// any of their revised, authored, or reviewed attributes.
-				List<GKInstance> parentPathways = (List<GKInstance>) newRLE.getReferers(ReactomeJavaConstants.hasEvent);
-				for (GKInstance parentPathway : parentPathways)
-				{
-					// Get all non-Reactome instance edits in the revised, authored, and reviewed
-					// attributes of the Pathway.
-					List<GKInstance> parentPathwayIEs = getNonReactomeEventIEs(parentPathway);
+		}
+		catch (SQLException e)
+		{
+			e.printStackTrace();
+		}
+		catch (InvalidAttributeException e)
+		{
+			// TODO Auto-generated catch block
+			e.printStackTrace();
+		}
+		catch (Exception e)
+		{
+			// TODO Auto-generated catch block
+			e.printStackTrace();
+		}
+	}
+
+	private static Map<GKInstance, Set<Map<GKInstance, GKInstance>>> processPreexistingRLEs(Map<GKInstance, Set<Map<GKInstance, GKInstance>>> pathway2Reactions2NewIEs, GKInstance currentRLE, GKInstance previousRLE) throws InvalidAttributeException, Exception
+	{
+		Map<GKInstance, Set<Map<GKInstance, GKInstance>>> pathwaysMap = pathway2Reactions2NewIEs;
+		// Get all reviewed, authored, and revised values from the current and previous
+		// version of the RlE.
+		List<GKInstance> currentRLEReviewedInstances = currentRLE.getAttributeValuesList(ReactomeJavaConstants.reviewed);
+		List<GKInstance> currentRLEAuthoredInstances = currentRLE.getAttributeValuesList(ReactomeJavaConstants.authored);
+		List<GKInstance> currentRLERevisedInstances = currentRLE.getAttributeValuesList(ReactomeJavaConstants.revised);
+		List<GKInstance> previousRLEReviewedInstances = previousRLE.getAttributeValuesList(ReactomeJavaConstants.reviewed);
+		List<GKInstance> previousRLEAuthoredInstances = previousRLE.getAttributeValuesList(ReactomeJavaConstants.authored);
+		List<GKInstance> previousRLERevisedInstances = previousRLE.getAttributeValuesList(ReactomeJavaConstants.revised);
```

Comment:
For all these names, I find it a bit confusing later on not knowing what type of instances they are.  Could each of these variables have their names end with "InstanceEdits" instead of "Instances"?

Reply:
This will probably get renamed later. These names came from Justin's original code.

Reply:
Again, why not now with the refactoring clean-up?

Reply:
I'm planning a significant overhaul that will be more object-oriented and I suspect these variables won't even make it (at least not in this state). So I'm reluctant to embark on so many renaming exercises when I suspect the future shape of this could be quite different and I have other things to do right now than renaming.